[Inspector] Use tabs instead of dropdown for view selection

matthewabbott opened the pull request:
## Summary

Closes https://github.com/elastic/kibana/issues/126918.

Currently, the inspector uses a button and dropdown for just two options as in this screenshot:

<img width="1402" alt="Screenshot 2025-03-03 at 1 45 35 AM" src="https://github.com/user-attachments/assets/92108828-fcbb-4dd6-bc2f-446b136c5060" />

---

This PR switches it to instead use tabs as in this screenshot:

<img width="1403" alt="Screenshot 2025-03-03 at 12 40 06 AM" src="https://github.com/user-attachments/assets/8da1d54f-3620-465f-80d6-89211e750815" />

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

matthewabbott committed:
Use tabs for inspector view selection

matthewabbott committed:
Update inspector panel test snapshot

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

elasticmachine commented:
Pinging @elastic/kibana-presentation (Team:Presentation)

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/280532)
* Commit: c2eb46dc32911de7f650151c79453cef80b9bf4e

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/d9c30722072ded6f77d76dfe4bbe43ea3def0176) | [after](https://github.com/elastic/kibana/commit/c2eb46dc32911de7f650151c79453cef80b9bf4e) | diff |
| --- | --- | --- | --- |
| `inspector` | 26.5KB | 26.6KB | +131.0B |



<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280532","number":280532,"commit":"c2eb46dc32911de7f650151c79453cef80b9bf4e"}],"number":280532}
buildkite-pr-comment-->

nreese commented on src/platform/plugins/shared/inspector/public/ui/inspector_view_chooser.tsx:
```diff
@@ -89,37 +83,37 @@ export class InspectorViewChooser extends Component<Props, State> {
       return this.renderSingleView();
     }
 
-    const triggerButton = this.renderViewButton();
```
`renderViewButton` method is no longer called and can be removed

nreese reviewed (commented):
Thanks for adding tabs.In addition to the changes, the functional testing code at https://github.com/elastic/kibana/blob/main/test/functional/services/inspector.ts#L191 should be updated. The `inspectorViewChooser` button no longer exists so there is no need to click it and instead, the tab can just be selected directly.

matthewabbott committed:
Remove now-unused button render method.

matthewabbott committed:
Update functional tests to use new tabs

matthewabbott commented:
> functional testing code at https://github.com/elastic/kibana/blob/main/test/functional/services/inspector.ts#L191 should be updated. The `inspectorViewChooser` button no longer exists so there is no need to click it and instead, the tab can just be selected directly.

Just tweaked the functional tests. Not 100% sure I correctly tested them locally. I got no failures when running:
```
node scripts/functional_test_runner --config test/functional/apps/home/config.ts
```

But running it with `--grep="inspector"` did no tests, so I’m not sure I was targeting the right config.
```
node scripts/functional_test_runner --config test/functional/apps/home/config.ts --grep="inspector"
```

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/280957)
* Commit: 8d0b1b11ced667ac067ce75bd7a134911ac5b38f

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/280957#019563ed-0a74-4dbe-8eaa-630741301315)

### History
* :green_heart: [Build #280532](https://buildkite.com/elastic/kibana-pull-request/builds/280532) succeeded c2eb46dc32911de7f650151c79453cef80b9bf4e

cc @cqliu1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280957","number":280957,"commit":"8d0b1b11ced667ac067ce75bd7a134911ac5b38f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280532","number":280532,"commit":"c2eb46dc32911de7f650151c79453cef80b9bf4e"}],"number":280957}
buildkite-pr-comment-->

matthewabbott committed:
Update tab to use the same i18n message ID the old button implementation was using

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/281263)
* Commit: df3cb618c7187874026f4aeda80b2d6c69a878bd

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/d9c30722072ded6f77d76dfe4bbe43ea3def0176) | [after](https://github.com/elastic/kibana/commit/df3cb618c7187874026f4aeda80b2d6c69a878bd) | diff |
| --- | --- | --- | --- |
| `inspector` | 26.5KB | 26.4KB | -69.0B |



### History
* :broken_heart: [Build #280957](https://buildkite.com/elastic/kibana-pull-request/builds/280957) failed 8d0b1b11ced667ac067ce75bd7a134911ac5b38f
* :green_heart: [Build #280532](https://buildkite.com/elastic/kibana-pull-request/builds/280532) succeeded c2eb46dc32911de7f650151c79453cef80b9bf4e

cc @cqliu1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281263","number":281263,"commit":"df3cb618c7187874026f4aeda80b2d6c69a878bd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280957","number":280957,"commit":"8d0b1b11ced667ac067ce75bd7a134911ac5b38f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280532","number":280532,"commit":"c2eb46dc32911de7f650151c79453cef80b9bf4e"}],"number":281263}
buildkite-pr-comment-->

matthewabbott commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into inspector/view-selection-tabs

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/281282)
* Commit: 05d6ccc39a1a33032458897bf635916089087004

### Failed CI Steps
* [FTR Configs #105](https://buildkite.com/elastic/kibana-pull-request/builds/281282#01956954-7b1a-47dc-be1b-869910df4489)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/281282#01956954-7b1a-47dc-be1b-869910df4489) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/281282/jobs/01956954-7b1a-47dc-be1b-869910df4489/artifacts/01956974-1886-4b95-a7a7-7bca45564119) FTR Configs #<span></span>105 / discover/group1 discover accessibility top nav menu buttons should return focus to the alerts button when dismissing the alerts popover

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/04ee5fc4f355dedba3d5c7e2183417c04a6dc5c0) | [after](https://github.com/elastic/kibana/commit/05d6ccc39a1a33032458897bf635916089087004) | diff |
| --- | --- | --- | --- |
| `inspector` | 23.4KB | 23.3KB | -167.0B |



### History
* :green_heart: [Build #281263](https://buildkite.com/elastic/kibana-pull-request/builds/281263) succeeded df3cb618c7187874026f4aeda80b2d6c69a878bd
* :broken_heart: [Build #280957](https://buildkite.com/elastic/kibana-pull-request/builds/280957) failed 8d0b1b11ced667ac067ce75bd7a134911ac5b38f
* :green_heart: [Build #280532](https://buildkite.com/elastic/kibana-pull-request/builds/280532) succeeded c2eb46dc32911de7f650151c79453cef80b9bf4e

cc @cqliu1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281282","number":281282,"commit":"05d6ccc39a1a33032458897bf635916089087004"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281263","number":281263,"commit":"df3cb618c7187874026f4aeda80b2d6c69a878bd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280957","number":280957,"commit":"8d0b1b11ced667ac067ce75bd7a134911ac5b38f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280532","number":280532,"commit":"c2eb46dc32911de7f650151c79453cef80b9bf4e"}],"number":281282}
buildkite-pr-comment-->

nreese commented on src/platform/plugins/shared/inspector/public/ui/inspector_panel.tsx:
```diff
@@ -134,15 +134,15 @@ export class InspectorPanel extends Component<InspectorPanelProps, InspectorPane
-            </EuiFlexItem>
           </EuiFlexGroup>
         </EuiFlyoutHeader>
+        <EuiFlexItem className="insInspectorViewChooserContainer">
```
Is `EuiFlexItem` still needed, since this is not longer in `EuiFlexGroup`. how about just using a `div` here.

nreese commented on src/platform/plugins/shared/inspector/public/ui/inspector_view_chooser.tsx:
```diff
@@ -69,57 +63,43 @@ export class InspectorViewChooser extends Component<Props, State> {
-
   renderSingleView() {
     return (
-      <EuiToolTip position="bottom" content={this.props.selectedView.help}>
```
`this.props.selectedView.help` is not longer displayed anywhere on the page. Can we keep this? Its useful information.

nreese reviewed (commented):
There are a few more test cases that need to be cleaned up
* [map inspect view chooser](https://github.com/elastic/kibana/blob/main/x-pack/test/accessibility/apps/group3/maps.ts#L60) - this case can be deleted
* [dashboard panel - inspector view chooser](https://github.com/elastic/kibana/blob/main/x-pack/test/accessibility/apps/group1/dashboard_panel_options.ts#L58) - this case can be deleted
* [openInspectorRequestsView](https://github.com/elastic/kibana/blob/main/test/functional/services/inspector.ts#L208) - remove check for `this.testSubjects.exists('inspectorViewChooser')`

matthewabbott committed:
Remove obsolete test cases

matthewabbott committed:
Replace unneeded euiflexitem with div

matthewabbott committed:
Restore EuiToolTips

matthewabbott committed:
Merge branch 'main' into inspector/view-selection-tabs

matthewabbott committed:
Remove check for inspector view chooser

matthewabbott commented:
Thanks @nreese! I removed the obsolete test cases, removed the check from that one case, and made the changes you suggested RE: flex item and restoring the tooltips. Really appreciate your help.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/283052)
* Commit: 0c015703ab0980c719f0411851f2789746b758e5

### Failed CI Steps
* [Jest Tests #9](https://buildkite.com/elastic/kibana-pull-request/builds/283052#019587e9-2d58-4eb5-9a2b-6035e6bc9547)
* [Jest Tests #9](https://buildkite.com/elastic/kibana-pull-request/builds/283052#01958809-9c31-402e-959a-e6528b743176)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/283052#019587e9-2d58-4eb5-9a2b-6035e6bc9547) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/283052/jobs/019587e9-2d58-4eb5-9a2b-6035e6bc9547/artifacts/01958809-64a8-4ab9-8e5f-b38abb3e61f8) Jest Tests #<span></span>9 / InspectorPanel should render as expected
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/283052#01958809-9c31-402e-959a-e6528b743176) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/283052/jobs/01958809-9c31-402e-959a-e6528b743176/artifacts/0195880e-3847-4e2d-bc01-e5ebc42602e9) Jest Tests #<span></span>9 / InspectorPanel should render as expected

### History
* :yellow_heart: [Build #281282](https://buildkite.com/elastic/kibana-pull-request/builds/281282) was flaky 05d6ccc39a1a33032458897bf635916089087004
* :green_heart: [Build #281263](https://buildkite.com/elastic/kibana-pull-request/builds/281263) succeeded df3cb618c7187874026f4aeda80b2d6c69a878bd
* :broken_heart: [Build #280957](https://buildkite.com/elastic/kibana-pull-request/builds/280957) failed 8d0b1b11ced667ac067ce75bd7a134911ac5b38f
* :green_heart: [Build #280532](https://buildkite.com/elastic/kibana-pull-request/builds/280532) succeeded c2eb46dc32911de7f650151c79453cef80b9bf4e

cc @cqliu1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283052","number":283052,"commit":"0c015703ab0980c719f0411851f2789746b758e5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281282","number":281282,"commit":"05d6ccc39a1a33032458897bf635916089087004"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281263","number":281263,"commit":"df3cb618c7187874026f4aeda80b2d6c69a878bd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280957","number":280957,"commit":"8d0b1b11ced667ac067ce75bd7a134911ac5b38f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280532","number":280532,"commit":"c2eb46dc32911de7f650151c79453cef80b9bf4e"}],"number":283052}
buildkite-pr-comment-->

cqliu1 committed:
Merge branch 'main' into inspector/view-selection-tabs

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/283066)
* Commit: 4c8c43bd2ac9dc6842ded8e3980c334300899106

### Failed CI Steps
* [Jest Tests #13](https://buildkite.com/elastic/kibana-pull-request/builds/283066#019588b3-0eb7-44a7-8243-cab1cc044689)
* [Jest Tests #13](https://buildkite.com/elastic/kibana-pull-request/builds/283066#019588d8-3754-4a40-8ef6-d79d482b7355)
* [FTR Configs #89](https://buildkite.com/elastic/kibana-pull-request/builds/283066#019588b3-0f0a-408e-9b5a-8195bf26cae4)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/283066#019588b3-0f0a-408e-9b5a-8195bf26cae4) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/283066/jobs/019588b3-0f0a-408e-9b5a-8195bf26cae4/artifacts/019588d1-60cd-4951-afa8-055db64170b4) FTR Configs #<span></span>89 / discover/group6 discover sidebar field stats should work for regular and pinned filters
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/283066#019588b3-0eb7-44a7-8243-cab1cc044689) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/283066/jobs/019588b3-0eb7-44a7-8243-cab1cc044689/artifacts/019588d8-0ed3-4979-b5a0-7ed60777725d) Jest Tests #<span></span>13 / InspectorPanel should render as expected
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/283066#019588d8-3754-4a40-8ef6-d79d482b7355) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/283066/jobs/019588d8-3754-4a40-8ef6-d79d482b7355/artifacts/019588dc-b132-44f4-80df-9ffdec42337b) Jest Tests #<span></span>13 / InspectorPanel should render as expected

### History
* :broken_heart: [Build #283052](https://buildkite.com/elastic/kibana-pull-request/builds/283052) failed 0c015703ab0980c719f0411851f2789746b758e5
* :yellow_heart: [Build #281282](https://buildkite.com/elastic/kibana-pull-request/builds/281282) was flaky 05d6ccc39a1a33032458897bf635916089087004
* :green_heart: [Build #281263](https://buildkite.com/elastic/kibana-pull-request/builds/281263) succeeded df3cb618c7187874026f4aeda80b2d6c69a878bd
* :broken_heart: [Build #280957](https://buildkite.com/elastic/kibana-pull-request/builds/280957) failed 8d0b1b11ced667ac067ce75bd7a134911ac5b38f
* :green_heart: [Build #280532](https://buildkite.com/elastic/kibana-pull-request/builds/280532) succeeded c2eb46dc32911de7f650151c79453cef80b9bf4e

cc @cqliu1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283066","number":283066,"commit":"4c8c43bd2ac9dc6842ded8e3980c334300899106"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283052","number":283052,"commit":"0c015703ab0980c719f0411851f2789746b758e5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281282","number":281282,"commit":"05d6ccc39a1a33032458897bf635916089087004"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281263","number":281263,"commit":"df3cb618c7187874026f4aeda80b2d6c69a878bd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280957","number":280957,"commit":"8d0b1b11ced667ac067ce75bd7a134911ac5b38f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280532","number":280532,"commit":"c2eb46dc32911de7f650151c79453cef80b9bf4e"}],"number":283066}
buildkite-pr-comment-->

matthewabbott committed:
Readjust test snapshot with new UI structure

nreese commented on src/platform/plugins/shared/inspector/public/ui/inspector_view_chooser.tsx:
```diff
@@ -69,57 +63,44 @@ export class InspectorViewChooser extends Component<Props, State> {
-
+    const { views, selectedView } = this.props;
     if (views.length < 2) {
       return this.renderSingleView();
```
I will defer to @elastic/kibana-design, but I think we should `return null;` for a single view instead of showing a tab with a single selection.

matthewabbott replied:
I went with the `defaultMessage="View: {viewName}"` --> `defaultMessage="{viewName}"` tweak rather than removing the `renderSingleView` method, but I’m happy to switch to null depending on if @elastic/kibana-design has a recommendation.

I was thinking since the single tab would still have that hover tooltip, that might be enough for it to be worth keeping around.

nreese commented on src/platform/plugins/shared/inspector/public/ui/inspector_view_chooser.tsx:
```diff
@@ -69,57 +63,44 @@ export class InspectorViewChooser extends Component<Props, State> {
+          >
+            <FormattedMessage
+              id="inspector.view"
+              defaultMessage="View: {viewName}"
```
If design wants to keep the single tag, how about just using `defaultMessage={viewName}`?

nreese reviewed (approved):
In https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/maps/public/inspector/vector_tile_adapter/components/requests_view_callout.tsx#L17, could you change this message to "You're viewing vector tile search requests. To view requests submitted to the search API, select Requests tab."

kibana-presentation changes LGTM
code review and tested in chrome

matthewabbott committed:
Merge branch 'main' into inspector/view-selection-tabs

matthewabbott committed:
Tweak request view callout message

matthewabbott committed:
Tweak single-view default message

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/284360)
* Commit: 90d5ad8baad7514c9439a4d7becbe2677b081d6d

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/b3010930f92df55dc39aaee6fb310f32f51215de) | [after](https://github.com/elastic/kibana/commit/90d5ad8baad7514c9439a4d7becbe2677b081d6d) | diff |
| --- | --- | --- | --- |
| `maps` | 3.0MB | 3.0MB | -1.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/b3010930f92df55dc39aaee6fb310f32f51215de) | [after](https://github.com/elastic/kibana/commit/90d5ad8baad7514c9439a4d7becbe2677b081d6d) | diff |
| --- | --- | --- | --- |
| `inspector` | 23.4KB | 23.4KB | -41.0B |



### History
* :broken_heart: [Build #283066](https://buildkite.com/elastic/kibana-pull-request/builds/283066) failed 4c8c43bd2ac9dc6842ded8e3980c334300899106
* :broken_heart: [Build #283052](https://buildkite.com/elastic/kibana-pull-request/builds/283052) failed 0c015703ab0980c719f0411851f2789746b758e5
* :yellow_heart: [Build #281282](https://buildkite.com/elastic/kibana-pull-request/builds/281282) was flaky 05d6ccc39a1a33032458897bf635916089087004
* :green_heart: [Build #281263](https://buildkite.com/elastic/kibana-pull-request/builds/281263) succeeded df3cb618c7187874026f4aeda80b2d6c69a878bd
* :broken_heart: [Build #280957](https://buildkite.com/elastic/kibana-pull-request/builds/280957) failed 8d0b1b11ced667ac067ce75bd7a134911ac5b38f

cc @cqliu1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/284360","number":284360,"commit":"90d5ad8baad7514c9439a4d7becbe2677b081d6d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283066","number":283066,"commit":"4c8c43bd2ac9dc6842ded8e3980c334300899106"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283052","number":283052,"commit":"0c015703ab0980c719f0411851f2789746b758e5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281282","number":281282,"commit":"05d6ccc39a1a33032458897bf635916089087004"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281263","number":281263,"commit":"df3cb618c7187874026f4aeda80b2d6c69a878bd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280957","number":280957,"commit":"8d0b1b11ced667ac067ce75bd7a134911ac5b38f"}],"number":284360}
buildkite-pr-comment-->

MichaelMarcialis commented on x-pack/platform/plugins/shared/maps/public/inspector/vector_tile_adapter/components/requests_view_callout.tsx:
```diff
@@ -14,7 +14,7 @@ export function RequestsViewCallout() {
     <EuiCallOut
       size="s"
       title={i18n.translate('xpack.maps.inspector.vectorTile.requestsView', {
-        defaultMessage: `You're viewing vector tile search requests. To view requests submitted to the search API, set View to Requests.`,
+        defaultMessage: `You're viewing vector tile search requests. To view requests submitted to the search API, select Requests tab.`,
       })}
       iconType="iInCircle"
     />
```
Under what conditions is this callout shown? The message strikes me as overly obvious, but perhaps I'm not understanding its intended purpose. Do we actually need this callout?

nreese replied:
Visualizations like lens show "Request" view by default. Maps shows the "Vector tile" view by default. They are different view for different type of requests. This callout is for maps vector tile inspector view.  Map data can be loaded from vector tile requests or search DSL requests. Users do not really know which to expect as it depends on the implementation of the layer.

MichaelMarcialis reviewed (commented):
Thanks for putting this together, @matthewabbott. Apologies for the delayed review (I was traveling for an off-site meeting). I've added a handful of comments/questions/suggestions for your initial review. Let me know if you have any questions.

nreese commented on src/platform/plugins/shared/inspector/public/ui/inspector_view_chooser.tsx:
```diff
@@ -69,57 +63,44 @@ export class InspectorViewChooser extends Component<Props, State> {
-      </EuiPopover>
+      <EuiTabs size="s" data-test-subj="inspectorViewChooser">
+        {views.map((view, index) => (
+          <EuiToolTip key={index} content={view.help} position="bottom">
+            <EuiTab
+              onClick={() => this.props.onViewSelected(view)}
+              isSelected={selectedView === view}
+              data-test-subj={`inspectorViewChooser${view.title}`}
+            >
+              {view.title}
+            </EuiTab>
+          </EuiToolTip>
```
I asked to keep the tooltip because the description is not the same as the title for different inspector views. the description is a registered field for each inspector view and should be accessable by the user.

MichaelMarcialis replied:
These are the tooltips I'm seeing currently:

- Data: View the data behind the visualization.
- Requests: View the search requests used to collect the data.

Are there others? If not, these two tooltips are the ones that I found to be obvious or redundant with the title.

nreese replied:
> Are there others? If not, these two tooltips are the ones that I found to be obvious or redundant with the title.

Yes. Open maps application and create a new map. Add a document layer. Now open inspector. Notice how there are 2 view, "Vector tiles" and "Requests". This is where the tooltips become important. Vector tiles are http requests but they are binary so they can not be tracked or viewed by the "Requests" adapter. Without tooltips, its more difficult to know what these are and why they are different.

<img width="500" alt="Screenshot 2025-04-04 at 8 48 48 AM" src="https://github.com/user-attachments/assets/458788e0-346f-4a03-b114-35c89dfbf9cd" />

MichaelMarcialis replied:
Thanks for that info, @nreese! Even still, I'm still a bit torn regarding the tooltips. But at the same time they're not actively doing harm, so I'll ultimately defer to you.

That said, I believe the entire inspector flyout could likely do with a bit of an IA refresh in general (maybe something along the lines of what we were planning as [part of the CCS enhancements](https://www.figma.com/design/AcDrAa4OzT4BTTu8xArjJG/Mockups---Version-4--In-Progress-?node-id=1021-221164&t=Totkel03G6rNlfqF-1) we did a while back), which might also help to alleviate the need for such tooltips. However that likely goes far beyond the scope of this PR.

In any case, I've recently transitioned away from the Analytics XP team. CCing @andreadelrio on this conversation in case she has any thoughts/opinions on the tooltips or the potential for a larger inspector IA review.
